### ISSUE TITLE
how to train on image with only one channel?

### ISSUE THREAD

**Issue by pkuyijia** [lucidrains/denoising-diffusion-pytorch#119]
I want to use the model to train with images which have only one channel, such as MNIST.
However, after I change `channels = 1` in Unet, there's still some size problems in `loss = diffusion(torch.randn(8, 1, 28, 28))`.
How can I solve it?

**Comment by lucidrains** [comment]
@pkuyijia :wave: 

```python
import torch
from denoising_diffusion_pytorch import Unet, GaussianDiffusion

model = Unet(
    dim = 64,
    dim_mults = (1, 2),
    channels = 1
)

diffusion = GaussianDiffusion(
    model,
    image_size = 28,
    timesteps = 1000,   # number of steps
)

training_images = torch.randn(8, 1, 28, 28) # images are normalized from 0 to 1
loss = diffusion(training_images)
loss.backward()
# after a lot of training

sampled_images = diffusion.sample(batch_size = 4)
sampled_images.shape # (4, 3, 128, 128)
```

**Comment by pkuyijia** [comment]
It works, thanks!

**Comment by RimeSea** [comment]
> @pkuyijia 👋
> 
> ```python
> import torch
> from denoising_diffusion_pytorch import Unet, GaussianDiffusion
> 
> model = Unet(
>     dim = 64,
>     dim_mults = (1, 2),
>     channels = 1
> )
> 
> diffusion = GaussianDiffusion(
>     model,
>     image_size = 28,
>     timesteps = 1000,   # number of steps
> )
> 
> training_images = torch.randn(8, 1, 28, 28) # images are normalized from 0 to 1
> loss = diffusion(training_images)
> loss.backward()
> # after a lot of training
> 
> sampled_images = diffusion.sample(batch_size = 4)
> sampled_images.shape # (4, 1, 28, 28)
> ```

Hello, may I ask what the principle behind this change is? Why can it be changed to [1,2] to work? Thank you very much.
